module/zfs/vdev.c

```diff
@@ -5636,6 +5636,31 @@ vdev_prop_set(vdev_t *vd, nvlist_t *innvl, nvlist_t *outnvl)
 	    innvl, 6, ZFS_SPACE_CHECK_EXTRA_RESERVED));
 }
 
+static int
+vdev_prop_get_int(vdev_t *vd, vdev_prop_t prop, uint64_t *value)
+{
+	spa_t *spa = vd->vdev_spa;
+	objset_t *mos = spa->spa_meta_objset;
+	uint64_t objid;
+	int err;
+
+	if (vd->vdev_top_zap != 0) {
+		objid = vd->vdev_top_zap;
+	} else if (vd->vdev_leaf_zap != 0) {
+		objid = vd->vdev_leaf_zap;
+	} else {
+		return (EINVAL);
```

Comment:
I was wondering if preferring the top over the leaf made sense or if there could be something that is neither top nor leaf, but this matches existing logic below in `vdev_prop_get()` except for the lack of `SET_ERROR()`.

Reply:
good question, I think the root vdev is neither top or leaf...and no zap. This function was partially factored out of `vdev_prop_get()` and I brought this sanity check over with it.

`vdev_prop_get()` is a bit imperfect as it doesn't handle available spares or cache vdevs correctly since they don't have a top_zap or a leaf_zap (as I understand it).